# Análisis de Estadísticas Vitales - EEVV

## Integrantes - Grupo 15

- Sara Sofia Cárdenas Rodríguez - 202214907
- Daniel Felipe Diaz Moreno - 202210773
- Juan Sebastián Urrea López - 201914710


## Los datos

Se trabaja con un conjunto de datos para cada uno de los 4 años requeridos.


## 1. Carga y Manipulación básica de los Datos

In [3]:
# Importaciones de bibliotecas
!pip  install xlrd pandas numpy
import pandas as pd
import numpy as np

  Using cached xlrd-2.0.1-py2.py3-none-any.whl.metadata (3.4 kB)
  Using cached pytz-2024.2-py2.py3-none-any.whl.metadata (22 kB)
  Using cached tzdata-2024.2-py2.py3-none-any.whl.metadata (1.4 kB)
Using cached xlrd-2.0.1-py2.py3-none-any.whl (96 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 11.3/11.3 MB 5.8 MB/s eta 0:00:00 0:00:01m
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.1/5.1 MB 31.5 MB/s eta 0:00:00
Using cached pytz-2024.2-py2.py3-none-any.whl (508 kB)
Using cached tzdata-2024.2-py2.py3-none-any.whl (346 kB)


In [4]:
nac2017 = pd.read_csv("nac2017.csv", delimiter=",", encoding="latin1")
nac2018 = pd.read_csv("nac2018.csv", delimiter=",", encoding="latin1")
nac2021 = pd.read_csv("nac2021.csv", delimiter=",", encoding="latin1")
nac2022 = pd.read_csv("nac2022.csv", delimiter=",", encoding="latin1")

print(f"Las dimensiones de nac2017 son: {nac2017.shape}")
print(f"Las dimensiones de nac2018 son: {nac2018.shape}")
print(f"Las dimensiones de nac2021 son: {nac2021.shape}")
print(f"Las dimensiones de nac2022 son: {nac2022.shape}")

Las dimensiones de nac2017 son: (656704, 38)
Las dimensiones de nac2018 son: (649115, 38)
Las dimensiones de nac2021 son: (616914, 39)
Las dimensiones de nac2022 son: (573625, 39)


/var/folders/9y/__pdhdy15bs2zh7r1m4dhnm40000gp/T/ipykernel_49041/1131811711.py:4: DtypeWarning: Columns (4) have mixed types. Specify dtype option on import or set low_memory=False.
  nac2022 = pd.read_csv("nac2022.csv", delimiter=",", encoding="latin1")


Las variables candidatas descritas en el informe son las siguientes:

- **Caracterización de la fecha**
  - `ANO`: Año de ocurrencia del nacimiento
  - `MES`: Mes de la ocurrencia del nacimiento

- **Caracterización de las regiones específicas**
  - `COD_DEPTO`: Departamento de nacimiento
  - `COD_MUNIC`: Municipio de nacimiento
  - `AREANAC`: Área de nacimiento (Cabecera municipal, centro poblado, rural disperso)

- **Factores socioeconómicos de la madre**
  - `CODPTORE`: Departamento de residencia habitual de la madre
  - `CODMUNRE`: Municipio de residencia habitual de la madre
  - `AREA_RES`: Área de residencia habitual de la madre
  - `EDAD_MADRE`: Edad de la madre a la fecha del parto (Rangos de edad)
  - `EST_CIVM`: Estado conyugal de la madre
  - `NIV_EDUM`: Último nivel de estudio que aprobó la madre
  - `ULTCURMAD`: Último año o grado aprobado de la madre
  - `SEG_SOCIAL`: Régimen de seguridad social en salud de la madre (Contributivo, subsidiado, excepción, especial, no asegurado)
  - `NUMCONSUL`: Número de consultas prenatales que tuvo la madre del nacido vivo

- **Caracterización clínica de la madre**
  - `N_HIJOSV`: Número de hijos nacidos vivos que ha tenido la madre, incluido el presente
  - `N_EMB`: Número de embarazos, incluido el presente

- **Caracterización del nacido vivo**
  - `SEXO`: Sexo del nacido vivo
  - `PESO_NAC`: Peso del nacido vivo, al nacer
  - `TALLA_NAC`: Talla del nacido vivo, al nacer
  - `T_GES`: Tiempo de gestación del nacido vivo (Menos de 22, de 22 a 27, de 28 a 37, de 38 a 41, de 42 y más)
  - `T_GES_AGRU_CIE`: Tiempo de gestación del nacido vivo ajustado a la agrupación sugerida por la CIE (Menos de 22, de 22 a 27, de 28 a 36, de 37 a 41, de 42 y más)

- **Caracterización del parto**
  - `TIPO_PARTO`: Tipo de parto de este nacimiento (Espontáneo, cesárea, instrumentado)
  - `MUL_PARTO`: Multiplicidad del embarazo (Simple, doble, triple, cuádruple o más)

- **Factores socioeconómicos del padre**
  - `EDAD_PADRE`: Edad del padre a la fecha del parto
  - `NIV_EDUP`: Nivel educativo del padre, último año de estudio que aprobó el padre
  - `ULTCURPAD`: Último año o grado aprobado del padre

- **Caracterización de la atención del parto**
  - `SIT_PARTO`: Sitio de parto (Institución de salud, domicilio)
  - `ATEN_PAR`: El parto fue atendido por (Médico, enfermero(a), auxiliar de enfermería, promotor(a) de salud, partera)
  - `PROFESION`: Profesión de quien certifica el nacimiento (Médico, enfermero(a), auxiliar de enfermería, promotor(a) de salud, funcionario de registro civil)

In [14]:
variablesABuscar = {
    # Caracterización de la fecha
    "ANO",  # Año de ocurrencia del nacimiento
    "MES",  # Mes de la ocurrencia del nacimiento
    
    # Caracterización de las regiones específicas
    "COD_DPTO",  # Departamento de nacimiento
    "COD_MUNIC",  # Municipio de nacimiento
    "AREANAC",  # Área de nacimiento (Cabecera municipal, centro poblado, rural disperso)
    
    # Factores socioeconómicos de la madre
    "CODPTORE",  # Departamento de residencia habitual de la madre
    "CODMUNRE",  # Municipio de residencia habitual de la madre
    "AREA_RES",  # Área de residencia habitual de la madre
    "EDAD_MADRE",  # Edad de la madre a la fecha del parto (Rangos de edad)
    "EST_CIVM",  # Estado conyugal de la madre
    "NIV_EDUM",  # Último nivel de estudio que aprobó la madre
    "ULTCURMAD",  # Último año o grado aprobado de la madre
    "SEG_SOCIAL",  # Régimen de seguridad social en salud de la madre
    "NUMCONSUL",  # Número de consultas prenatales que tuvo la madre del nacido vivo
    
    # Caracterización clínica de la madre
    "N_HIJOSV",  # Número de hijos nacidos vivos que ha tenido la madre, incluido el presente
    "N_EMB",  # Número de embarazos, incluido el presente
    
    # Caracterización del nacido vivo
    "SEXO",  # Sexo del nacido vivo
    "PESO_NAC",  # Peso del nacido vivo, al nacer
    "TALLA_NAC",  # Talla del nacido vivo, al nacer
    "T_GES",  # Tiempo de gestación del nacido vivo (Menos de 22, de 22 a 27, de 28 a 37, de 38 a 41, de 42 y más)
    "T_GES_AGRU_CIE",  # Tiempo de gestación del nacido vivo ajustado a la agrupación sugerida por la CIE
    
    # Caracterización del parto
    "TIPO_PARTO",  # Tipo de parto de este nacimiento
    "MUL_PARTO",  # Multiplicidad del embarazo (Simple, doble, triple, cuádruple o más)
    
    # Factores socioeconómicos del padre
    "EDAD_PADRE",  # Edad del padre a la fecha del parto
    "NIV_EDUP",  # Nivel educativo del padre, último año de estudio que aprobó el padre
    "ULTCURPAD",  # Último año o grado aprobado del padre
    
    # Caracterización de la atención del parto
    "SIT_PARTO",  # Sitio de parto (Institución de salud, domicilio)
    "ATEN_PAR",  # El parto fue atendido por
    "PROFESION"  # Profesión de quien certifica el nacimiento
}


columnasNac2017 = [col for col in nac2017.columns if col in variablesABuscar]
columnasNac2018 = [col for col in nac2018.columns if col in variablesABuscar]
columnasNac2021 = [col for col in nac2021.columns if col in variablesABuscar]
columnasNac2022 = [col for col in nac2022.columns if col in variablesABuscar]

columnasNac2017.sort()
columnasNac2018.sort()
columnasNac2021.sort()
columnasNac2022.sort()

print(f"Las columnas de salud2017 son: {columnasNac2017}")
print(f"Las columnas de salud2018 son: {columnasNac2018}")
print(f"Las columnas de salud2021 son: {columnasNac2021}")
print(f"Las columnas de salud2022 son: {columnasNac2022}")

columnasEnComun = set(columnasNac2017).intersection(set(columnasNac2018)).intersection(set(columnasNac2021)).intersection(set(columnasNac2022))

print(f"Las columnas en común son: {columnasEnComun}")
print(f"Las columnas que no estan en todos los datasets son: {variablesABuscar - columnasEnComun}")

Las columnas de salud2017 son: ['ANO', 'AREANAC', 'AREA_RES', 'ATEN_PAR', 'CODMUNRE', 'CODPTORE', 'COD_DPTO', 'COD_MUNIC', 'EDAD_MADRE', 'EDAD_PADRE', 'EST_CIVM', 'MES', 'MUL_PARTO', 'NIV_EDUM', 'NIV_EDUP', 'NUMCONSUL', 'N_EMB', 'N_HIJOSV', 'PESO_NAC', 'PROFESION', 'SEG_SOCIAL', 'SEXO', 'SIT_PARTO', 'TALLA_NAC', 'TIPO_PARTO', 'T_GES', 'T_GES_AGRU_CIE', 'ULTCURMAD', 'ULTCURPAD']
Las columnas de salud2018 son: ['ANO', 'AREANAC', 'AREA_RES', 'ATEN_PAR', 'CODMUNRE', 'CODPTORE', 'COD_DPTO', 'COD_MUNIC', 'EDAD_MADRE', 'EDAD_PADRE', 'EST_CIVM', 'MES', 'MUL_PARTO', 'NIV_EDUM', 'NIV_EDUP', 'NUMCONSUL', 'N_EMB', 'N_HIJOSV', 'PESO_NAC', 'PROFESION', 'SEG_SOCIAL', 'SEXO', 'SIT_PARTO', 'TALLA_NAC', 'TIPO_PARTO', 'T_GES', 'T_GES_AGRU_CIE', 'ULTCURMAD', 'ULTCURPAD']
Las columnas de salud2021 son: ['ANO', 'AREANAC', 'AREA_RES', 'ATEN_PAR', 'CODMUNRE', 'CODPTORE', 'COD_DPTO', 'COD_MUNIC', 'EDAD_MADRE', 'EDAD_PADRE', 'EST_CIVM', 'MES', 'MUL_PARTO', 'NIV_EDUM', 'NIV_EDUP', 'NUMCONSUL', 'N_EMB', 'N_HIJOSV

Se aprecia que las variables de interés se encuentran en los datasets de todos los años.

Con esto, a continuación se presenta una muestra para cada uno de los archivos de datos disponibles.

In [15]:
nac2017[columnasNac2017].sample(10)

ANO  AREANAC  AREA_RES  ATEN_PAR  CODMUNRE  CODPTORE  COD_DPTO  \
650467  2017        1       1.0         1     466.0      23.0        23   
645663  2017        1       1.0         1       1.0      27.0        27   
554099  2017        1       1.0         1       1.0      63.0        63   
495499  2017        1       1.0         1       1.0      23.0        23   
412695  2017        1       1.0         1       1.0      47.0        47   
459469  2017        1       2.0         1     672.0      23.0        23   
30377   2017        1       3.0         1     244.0      13.0        13   
345715  2017        1       1.0         1       1.0      11.0        11   
652065  2017        1       1.0         1     836.0      13.0        13   
276567  2017        1       1.0         1     430.0      13.0        13   

        COD_MUNIC  EDAD_MADRE  EDAD_PADRE  ...  PROFESION  SEG_SOCIAL  SEXO  \
650467        466           3          28  ...          1           2     2   
645663          1           5          33  ...          1           2     2   
554099          1           5          25  ...          1           2     2   
495499          1           4          28  ...          1           1     1   
412695          1           2          19  ...          1           1     2   
459469          1           3          25  ...          1           1     1   
30377         244           2          26  ...          1           2     1   
345715          1           4          31  ...          1           1     2   
652065          1           2          24  ...          1           2     1   
276567        430           4          31  ...          1           1     2   

        SIT_PARTO  TALLA_NAC  TIPO_PARTO  T_GES  T_GES_AGRU_CIE  ULTCURMAD  \
650467          1          4           1      4               4         11   
645663          1          4           2      4               4         11   
554099          1          4           1      3               4         11   
495499          1          5           2      4               4         11   
412695          1          4           1      3               4          9   
459469          1          4           2      3               4          2   
30377           1          3           1      3               3          1   
345715          1          5           1      4               4          3   
652065          1          4           2      4               4          9   
276567          1          5           2      4               4         99   

        ULTCURPAD  
650467         11  
645663         11  
554099         11  
495499         11  
412695          9  
459469         11  
30377           9  
345715          5  
652065          9  
276567         99  

[10 rows x 29 columns]

In [16]:
nac2018[columnasNac2018].sample(10)

ANO  AREANAC  AREA_RES  ATEN_PAR  CODMUNRE  CODPTORE  COD_DPTO  \
306540  2018        1       1.0         1       1.0      11.0        11   
285122  2018        1       1.0         1     396.0      41.0        41   
11382   2018        1       1.0         1       1.0      11.0        11   
291714  2018        1       1.0         1       1.0      11.0        11   
369345  2018        1       3.0         1     847.0      44.0        44   
590505  2018        1       1.0         1     594.0      63.0        63   
449145  2018        1       1.0         1       1.0      11.0        11   
495414  2018        1       1.0         1       1.0       8.0         8   
587413  2018        1       1.0         1       1.0      99.0        99   
293063  2018        1       1.0         1       1.0      76.0        76   

        COD_MUNIC  EDAD_MADRE  EDAD_PADRE  ...  PROFESION  SEG_SOCIAL  SEXO  \
306540          1           2          27  ...          1           1     1   
285122        396           3          25  ...          1           2     2   
11382           1           3          25  ...          1           1     2   
291714          1           4          28  ...          1           1     2   
369345        847           5         999  ...          1           2     1   
590505        130           6          39  ...          1           2     1   
449145          1           5          36  ...          1           1     1   
495414          1           6          37  ...          1           1     2   
587413          1           3          24  ...          1           2     1   
293063          1           4          31  ...          1           1     1   

        SIT_PARTO  TALLA_NAC  TIPO_PARTO  T_GES  T_GES_AGRU_CIE  ULTCURMAD  \
306540          1          5           1      4               4          9   
285122          1          4           2      4               4         11   
11382           1          4           1      4               4          2   
291714          1          5           2      4               4         11   
369345          1          5           1      4               4         11   
590505          1          5           2      4               4          2   
449145          1          5           2      4               4          3   
495414          1          4           2      4               4          2   
587413          1          5           2      4               4         11   
293063          1          5           2      3               4          5   

        ULTCURPAD  
306540          9  
285122         11  
11382          11  
291714         11  
369345         99  
590505          2  
449145          2  
495414          2  
587413          3  
293063          5  

[10 rows x 29 columns]

In [17]:
nac2021[columnasNac2021].sample(10)

ANO  AREANAC  AREA_RES  ATEN_PAR  CODMUNRE  CODPTORE  COD_DPTO  \
104130  2021        1       1.0         1       1.0      13.0        13   
599495  2021        1       1.0         1     541.0       5.0         5   
132922  2021        1       2.0         1     770.0      20.0        20   
307291  2021        1       1.0         1       1.0      11.0        11   
412078  2021        1       2.0         1     809.0      19.0        76   
90386   2021        2       2.0         1     980.0      47.0        47   
214168  2021        1       1.0         1       1.0      50.0        50   
313773  2021        1       1.0         1     754.0      25.0        11   
533027  2021        1       1.0         1     380.0       5.0         5   
93898   2021        1       1.0         1     430.0      44.0        44   

        COD_MUNIC  EDAD_MADRE  EDAD_PADRE  ...  PROFESION  SEG_SOCIAL  SEXO  \
104130          1           5          29  ...        1.0           1     1   
599495          1           2          23  ...        1.0           2     2   
132922          1           2          37  ...        1.0           2     1   
307291          1           7          33  ...        1.0           1     2   
412078          1           4          40  ...        1.0           1     1   
90386         980           3          25  ...        1.0           1     2   
214168          1           2          27  ...        1.0           1     1   
313773          1           4          30  ...        1.0           2     2   
533027        266           5          33  ...        1.0           2     2   
93898         430           5          33  ...        1.0           2     2   

        SIT_PARTO  TALLA_NAC  TIPO_PARTO  T_GES  T_GES_AGRU_CIE  ULTCURMAD  \
104130          1          4           2      3               4         11   
599495          1          4           1      4               4          9   
132922          1          5           1      4               4          9   
307291          1          4           1      4               4         11   
412078          1          5           1      4               4         11   
90386           1          5           1      4               4          9   
214168          1          5           2      4               4          5   
313773          1          5           1      4               4          2   
533027          1          5           2      4               4          9   
93898           1          5           2      4               4          5   

        ULTCURPAD  
104130         11  
599495          9  
132922         99  
307291         99  
412078         11  
90386           9  
214168         11  
313773          2  
533027          9  
93898          99  

[10 rows x 29 columns]

In [18]:
nac2022[columnasNac2022].sample(10)

ANO  AREANAC  AREA_RES  ATEN_PAR  CODMUNRE  CODPTORE  COD_DPTO  \
88242   2022        1       1.0         1     368.0       5.0         5   
511816  2022        1       1.0         1     843.0      25.0        25   
467712  2022        1       1.0         1       1.0      50.0        50   
309872  2022        1       1.0         1      78.0      44.0        44   
302455  2022        1       1.0         1       1.0       5.0         5   
446355  2022        1       1.0         1       1.0      27.0         5   
521081  2022        1       1.0         1     318.0      76.0        76   
540543  2022        1       1.0         1     572.0      68.0        15   
542351  2022        1       1.0         1     631.0       5.0         5   
144899  2022        1       1.0         1      58.0      47.0        20   

        COD_MUNIC  EDAD_MADRE  EDAD_PADRE  ...  PROFESION  SEG_SOCIAL  SEXO  \
88242           1           6          32  ...        1.0           1     1   
511816        843           5          42  ...        1.0           1     1   
467712          1           2          20  ...        1.0           1     1   
309872        650           4          31  ...        1.0           2     1   
302455          1           3          26  ...        1.0           2     1   
446355          1           5          28  ...        1.0           2     2   
521081        111           3          26  ...        1.0           1     2   
540543        469           3          33  ...        1.0           2     2   
542351          1           5          50  ...        1.0           1     2   
144899         60           6          33  ...        1.0           2     1   

        SIT_PARTO  TALLA_NAC  TIPO_PARTO  T_GES  T_GES_AGRU_CIE  ULTCURMAD  \
88242           1          4           2      3               4         11   
511816          1          5           2      4               4          3   
467712          1          4           1      4               4          6   
309872          1          5           2      4               4          2   
302455          1          4           2      4               4          5   
446355          1          4           2      3               3          2   
521081          1          4           1      4               4         11   
540543          1          5           1      4               4          5   
542351          1          4           1      3               3          5   
144899          1          5           2      3               4          5   

        ULTCURPAD  
88242          11  
511816          5  
467712          6  
309872          5  
302455         99  
446355         11  
521081         11  
540543          5  
542351         11  
144899         11  

[10 rows x 29 columns]

Posteriormente, se muestra el tipo de dato para cada atributo por cada archivo de datos.

In [19]:
nac2017[columnasNac2017].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 656704 entries, 0 to 656703
Data columns (total 29 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   ANO             656704 non-null  int64  
 1   AREANAC         656704 non-null  int64  
 2   AREA_RES        655508 non-null  float64
 3   ATEN_PAR        656704 non-null  int64  
 4   CODMUNRE        655437 non-null  float64
 5   CODPTORE        655438 non-null  float64
 6   COD_DPTO        656704 non-null  int64  
 7   COD_MUNIC       656704 non-null  int64  
 8   EDAD_MADRE      656704 non-null  int64  
 9   EDAD_PADRE      656704 non-null  int64  
 10  EST_CIVM        656704 non-null  int64  
 11  MES             656704 non-null  int64  
 12  MUL_PARTO       656704 non-null  int64  
 13  NIV_EDUM        656704 non-null  int64  
 14  NIV_EDUP        656704 non-null  int64  
 15  NUMCONSUL       656704 non-null  int64  
 16  N_EMB           656704 non-null  int64  
 17  N_HIJOSV  

In [20]:
nac2017[columnasNac2017].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 656704 entries, 0 to 656703
Data columns (total 29 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   ANO             656704 non-null  int64  
 1   AREANAC         656704 non-null  int64  
 2   AREA_RES        655508 non-null  float64
 3   ATEN_PAR        656704 non-null  int64  
 4   CODMUNRE        655437 non-null  float64
 5   CODPTORE        655438 non-null  float64
 6   COD_DPTO        656704 non-null  int64  
 7   COD_MUNIC       656704 non-null  int64  
 8   EDAD_MADRE      656704 non-null  int64  
 9   EDAD_PADRE      656704 non-null  int64  
 10  EST_CIVM        656704 non-null  int64  
 11  MES             656704 non-null  int64  
 12  MUL_PARTO       656704 non-null  int64  
 13  NIV_EDUM        656704 non-null  int64  
 14  NIV_EDUP        656704 non-null  int64  
 15  NUMCONSUL       656704 non-null  int64  
 16  N_EMB           656704 non-null  int64  
 17  N_HIJOSV  

In [21]:
nac2021[columnasNac2021].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 616914 entries, 0 to 616913
Data columns (total 29 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   ANO             616914 non-null  int64  
 1   AREANAC         616914 non-null  int64  
 2   AREA_RES        608086 non-null  float64
 3   ATEN_PAR        616914 non-null  int64  
 4   CODMUNRE        608083 non-null  float64
 5   CODPTORE        608083 non-null  float64
 6   COD_DPTO        616914 non-null  int64  
 7   COD_MUNIC       616914 non-null  int64  
 8   EDAD_MADRE      616914 non-null  int64  
 9   EDAD_PADRE      616914 non-null  int64  
 10  EST_CIVM        616914 non-null  int64  
 11  MES             616914 non-null  int64  
 12  MUL_PARTO       616914 non-null  int64  
 13  NIV_EDUM        616914 non-null  int64  
 14  NIV_EDUP        616914 non-null  int64  
 15  NUMCONSUL       616914 non-null  int64  
 16  N_EMB           616914 non-null  int64  
 17  N_HIJOSV  

In [22]:
nac2022[columnasNac2022].info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 573625 entries, 0 to 573624
Data columns (total 29 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   ANO             573625 non-null  int64  
 1   AREANAC         573625 non-null  int64  
 2   AREA_RES        567227 non-null  float64
 3   ATEN_PAR        573625 non-null  int64  
 4   CODMUNRE        567227 non-null  float64
 5   CODPTORE        567227 non-null  float64
 6   COD_DPTO        573625 non-null  int64  
 7   COD_MUNIC       573625 non-null  int64  
 8   EDAD_MADRE      573625 non-null  int64  
 9   EDAD_PADRE      573625 non-null  int64  
 10  EST_CIVM        573625 non-null  int64  
 11  MES             573625 non-null  int64  
 12  MUL_PARTO       573625 non-null  int64  
 13  NIV_EDUM        573625 non-null  int64  
 14  NIV_EDUP        573625 non-null  int64  
 15  NUMCONSUL       573625 non-null  int64  
 16  N_EMB           573625 non-null  int64  
 17  N_HIJOSV  

Finalmente, se unen los 4 datasets en uno unico, para facilitar el analisis de los datos

In [24]:
nac_df = pd.concat([nac2017[columnasNac2017], nac2018[columnasNac2018], nac2021[columnasNac2021], nac2022[columnasNac2022]], axis=0)
nac_df.reset_index(drop=True, inplace=True)
nac_df.sample(30)

ANO  AREANAC  AREA_RES  ATEN_PAR  CODMUNRE  CODPTORE  COD_DPTO  \
156242   2017        1       1.0         1      45.0       5.0         5   
662786   2018        1       1.0         1       1.0      54.0        54   
1014342  2018        1       2.0         1     417.0      23.0        23   
10349    2017        1       3.0         1     835.0      52.0        76   
2460344  2022        1       1.0         1       1.0      66.0        66   
736159   2018        1       1.0         1       1.0      11.0        11   
843931   2018        1       1.0         1       1.0      11.0        11   
2383247  2022        1       3.0         1     560.0      44.0        44   
688835   2018        1       1.0         1     837.0       5.0         5   
1070768  2018        1       1.0         1       1.0      85.0        85   
523204   2017        1       1.0         1       1.0      70.0        70   
1464611  2021        1       1.0         1       1.0      70.0        70   
2326673  2022        1       1.0         1     268.0      73.0        73   
1500795  2021        1       1.0         1       1.0      41.0        41   
159364   2017        1       3.0         1     755.0      68.0        68   
513435   2017        1       1.0         1       1.0      70.0        70   
153913   2017        1       1.0         1     740.0      25.0        11   
1829971  2021        1       2.0         1     837.0       5.0         5   
475554   2017        1       1.0         1     895.0       5.0         5   
2129816  2022        1       1.0         1     356.0      52.0        52   
2422729  2022        1       3.0         1     151.0      25.0        25   
1895294  2021        1       1.0         1     154.0       5.0         5   
1078496  2018        1       1.0         1       1.0      11.0        11   
1550380  2021        1       1.0         1       1.0      11.0        11   
707825   2018        1       1.0         1     754.0      25.0        11   
291089   2017        1       3.0         1     847.0      44.0        44   
35458    2017        1       1.0         1     754.0      25.0        11   
1471690  2021        1       1.0         1     279.0      44.0        44   
130808   2017        1       1.0         1     318.0      76.0        76   
831710   2018        1       3.0         1     547.0      68.0        68   

         COD_MUNIC  EDAD_MADRE  EDAD_PADRE  ...  PROFESION  SEG_SOCIAL  SEXO  \
156242          45           5          33  ...        1.0           1     2   
662786           1           4          38  ...        1.0           3     1   
1014342        417           3          30  ...        1.0           2     2   
10349            1           4          42  ...        1.0           1     1   
2460344          1           5          30  ...        1.0           1     1   
736159           1           3          26  ...        1.0           1     2   
843931           1           3          21  ...        1.0           1     1   
2383247          1           4          30  ...        1.0           1     2   
688835         837           6          40  ...        1.0           1     1   
1070768          1           3          28  ...        1.0           1     1   
523204           1           4          27  ...        1.0           2     1   
1464611          1           3          30  ...        1.0           5     2   
2326673        268           5          35  ...        1.0           2     2   
1500795          1           2          22  ...        1.0           2     1   
159364         755           3          31  ...        1.0           2     1   
513435           1           2          27  ...        1.0           1     1   
153913           1           2          22  ...        1.0           1     2   
1829971         45           6          40  ...        1.0           1     2   
475554         154           5          43  ...        1.0           2     1   
2129816        356           5          30  ...      

## 2. Entendimiento de los datos

## 2.1. Análisis de calidad de datos





## 2.1.1. Análisis de completitud
A continuación, se presentan los porcentajes de completitud e incompletitud para cada uno de los atributos.

In [25]:
porcentajeIncompletitudAtributo =  (100 * nac_df.isnull().sum() / nac_df.shape[0]).sort_values(ascending=False)
pd.DataFrame({
    "Atributo": porcentajeIncompletitudAtributo.index,
    "Completitud (%)": 100 - porcentajeIncompletitudAtributo.values,
    "Incompletitud (%)": porcentajeIncompletitudAtributo.values
})

Atributo  Completitud (%)  Incompletitud (%)
0         CODMUNRE        99.175319           0.824681
1         CODPTORE        99.175359           0.824641
2         AREA_RES        99.182449           0.817551
3        PROFESION        99.909228           0.090772
4              ANO       100.000000           0.000000
5            N_EMB       100.000000           0.000000
6        ULTCURMAD       100.000000           0.000000
7   T_GES_AGRU_CIE       100.000000           0.000000
8            T_GES       100.000000           0.000000
9       TIPO_PARTO       100.000000           0.000000
10       TALLA_NAC       100.000000           0.000000
11       SIT_PARTO       100.000000           0.000000
12            SEXO       100.000000           0.000000
13      SEG_SOCIAL       100.000000           0.000000
14        PESO_NAC       100.000000           0.000000
15        N_HIJOSV       100.000000           0.000000
16        NIV_EDUP       100.000000           0.000000
17       NUMCONSUL       100.000000           0.000000
18         AREANAC       100.000000           0.000000
19        NIV_EDUM       100.000000           0.000000
20       MUL_PARTO       100.000000           0.000000
21             MES       100.000000           0.000000
22        EST_CIVM       100.000000           0.000000
23      EDAD_PADRE       100.000000           0.000000
24      EDAD_MADRE       100.000000           0.000000
25       COD_MUNIC       100.000000           0.000000
26        COD_DPTO       100.000000           0.000000
27        ATEN_PAR       100.000000           0.000000
28       ULTCURPAD       100.000000           0.000000

Se puede ver que el atributo más incompleto es CODMUNRE, con apenas el 0.82% de datos faltantes.

## 2.1.2. Análisis de unicidad
A continuación, se buscará determinar cuántas filas duplicadas hay en el dataset

In [26]:
registrosDuplicados = nac_df.duplicated()
totalDuplicados = registrosDuplicados.sum()
totalOpiniones = nac_df.shape[0]
porcentajeDuplicados = 100.0 * totalDuplicados / totalOpiniones
print(f"Se cuenta con {totalDuplicados} registros duplicados ({porcentajeDuplicados:.2f}%)")



Se cuenta con 8387 registros duplicados (0.34%)


Los datos presentan un porcentaje relativamente pequeño de duplicidad, por lo que seria razonable eliminar dichos duplicados.

## 2.1.3. Análisis de validez








### **Caracterización de la fecha**
- **`ANO`**: Año de ocurrencia del nacimiento
  - Valores posibles: 2017, 2018, 2021, 2022

- **`MES`**: Mes de la ocurrencia del nacimiento
  - 01 = Enero
  - 02 = Febrero
  - 03 = Marzo
  - 04 = Abril
  - 05 = Mayo
  - 06 = Junio
  - 07 = Julio
  - 08 = Agosto
  - 09 = Septiembre
  - 10 = Octubre
  - 11 = Noviembre
  - 12 = Diciembre

### **Caracterización de las regiones específicas**
- **`COD_DEPTO`**: Departamento de nacimiento (según la División Político-Administrativa del DANE)
  - 05 = Antioquia
  - 08 = Atlántico
  - 11 = Bogotá
  - 13 = Bolívar
  - 15 = Boyacá
  - 17 = Caldas
  - 18 = Caquetá
  - 19 = Cauca
  - 20 = Cesar
  - 23 = Córdoba
  - 25 = Cundinamarca
  - 27 = Chocó
  - 41 = Huila
  - 44 = La Guajira
  - 47 = Magdalena
  - 50 = Meta
  - 52 = Nariño
  - 54 = Norte de Santander
  - 63 = Quindío
  - 66 = Risaralda
  - 68 = Santander
  - 70 = Sucre
  - 73 = Tolima
  - 76 = Valle del Cauca
  - 81 = Arauca
  - 85 = Casanare
  - 86 = Putumayo
  - 88 = Archipiélago de San Andrés, Providencia y Santa Catalina
  - 91 = Amazonas
  - 94 = Guainía
  - 95 = Guaviare
  - 97 = Vaupés
  - 99 = Vichada

- **`COD_MUNIC`**: Municipio de nacimiento (según la División Político-Administrativa del DANE)

- **`AREANAC`**: Área de nacimiento
  - 1 = Cabecera municipal
  - 2 = Centro poblado (Inspección, corregimiento o caserío)
  - 3 = Rural disperso
  - 9 = Sin información

### **Factores socioeconómicos de la madre**
- **`CODPTORE`**: Departamento de residencia habitual de la madre (según la División Político-Administrativa del DANE)

- **`CODMUNRE`**: Municipio de residencia habitual de la madre (según la División Político-Administrativa del DANE)

- **`AREA_RES`**: Área de residencia habitual de la madre
  - 1 = Cabecera municipal
  - 2 = Centro poblado (Inspección, corregimiento o caserío)
  - 3 = Rural disperso
  - 9 = Sin información

- **`EDAD_MADRE`**: Edad de la madre a la fecha del parto (en rangos de edad)
  - 1 = De 10-14 Años
  - 2 = De 15-19 Años
  - 3 = De 20-24 Años
  - 4 = De 25-29 Años
  - 5 = De 30-34 Años
  - 6 = De 35-39 Años
  - 7 = De 40-44 Años
  - 8 = De 45-49 Años
  - 9 = De 50-54 Años
  - 99 = Sin información

- **`EST_CIVM`**: Estado conyugal de la madre
  - 1 = No está casada y lleva dos o más años viviendo con su pareja
  - 2 = No está casada y lleva menos de dos años viviendo con su pareja
  - 3 = Está separada, divorciada
  - 4 = Está viuda
  - 5 = Está soltera
  - 6 = Está casada
  - 9 = Sin información

- **`NIV_EDUM`**: Último nivel de estudio que aprobó la madre
  - 1 = Preescolar
  - 2 = Básica primaria
  - 3 = Básica secundaria
  - 4 = Media académica o clásica
  - 5 = Media técnica
  - 6 = Normalista
  - 7 = Técnica profesional
  - 8 = Tecnológica
  - 9 = Profesional
  - 10 = Especialización
  - 11 = Maestría
  - 12 = Doctorado
  - 13 = Ninguno
  - 99 = Sin información

- **`ULTCURMAD`**: Último año o grado aprobado de la madre
  - Valores coinciden con `NIV_EDUM`

- **`SEG_SOCIAL`**: Régimen de seguridad social en salud de la madre
  - 1 = Contributivo
  - 2 = Subsidiado
  - 3 = Excepción
  - 4 = Especial
  - 5 = No asegurado
  - 9 = Sin información

- **`NUMCONSUL`**: Número de consultas prenatales que tuvo la madre del nacido vivo
  - 00 = Ninguna
  - 99 = Sin información

### **Caracterización clínica de la madre**
- **`N_HIJOSV`**: Número de hijos nacidos vivos que ha tenido la madre, incluido el presente
  - 0 a 20
  - 99 = Sin información

- **`N_EMB`**: Número de embarazos, incluido el presente

### **Caracterización del nacido vivo**
- **`SEXO`**: Sexo del nacido vivo
  - 1 = Masculino
  - 2 = Femenino
  - 3 = Indeterminado

- **`PESO_NAC`**: Peso del nacido vivo, al nacer
  - 1 = Menos de 1000 gramos
  - 2 = De 1000 a 1499 gramos
  - 3 = De 1500 a 1999 gramos
  - 4 = De 2000 a 2499 gramos
  - 5 = De 2500 a 2999 gramos
  - 6 = De 3000 a 3499 gramos
  - 7 = De 3500 a 3999 gramos
  - 8 = De 4000 y más gramos
  - 9 = Sin información

- **`TALLA_NAC`**: Talla del nacido vivo, al nacer
  - 1 = Menos de 20 cm
  - 2 = 20-29 cm
  - 3 = 30-39 cm
  - 4 = 40-49 cm
  - 5 = 50-59 cm
  - 6 = 60 cm o más
  - 9 = Sin información

- **`T_GES`**: Tiempo de gestación del nacido vivo
  - 1 = Menos de 22 semanas
  - 2 = De 22 a 27 semanas
  - 3 = De 28 a 37 semanas
  - 4 = De 38 a 41 semanas
  - 5 = 42 semanas o más
  - 6 = Ignorado
  - 9 = Sin información

- **`T_GES_AGRU_CIE`**: Tiempo de gestación del nacido vivo ajustado a la agrupación sugerida por la CIE
  - Valores similares a `T_GES`

### **Caracterización del parto**
- **`TIPO_PARTO`**: Tipo de parto de este nacimiento
  - 1 = Espontáneo
  - 2 = Cesárea
  - 3 = Instrumentado
  - 4 = Ignorado
  - 9 = Sin información

- **`MUL_PARTO`**: Multiplicidad del embarazo
  - 1 = Simple
  - 2 = Doble
  - 3 = Triple
  - 4 = Cuádruple o más
  - 9 = Sin información

### **Factores socioeconómicos del padre**
- **`EDAD_PADRE`**: Edad del padre a la fecha del parto, o 999 si no se conoce

- **`NIV_EDUP`**: Nivel educativo del padre
  - Valores coinciden con `NIV_EDUM`

- **`ULTCURPAD`**: Último año o grado aprobado del padre
  - Valores coinciden con `NIV_EDUM`

### **Caracterización de la atención del parto**
- **`SIT_PARTO`**: Sitio de parto
  - 1 = Institución de salud
  - 2 = Domicilio
  - 3 = Otro
  - 9 = Sin información

- **`ATEN_PAR`**: El parto fue atendido por
  - 1 = Médico
  - 2 = Enfermero(a)
  - 3 = Auxiliar de enfermería
  - 4 = Promotor(a) de salud
  - 5 = Partera
  - 6 = Otra persona
  - 9 = Sin información

- **`PROFESION`**: Profesión de quien certifica el nacimiento
  - 1 = Médico
  - 2 = Enfermero(a)
  - 3 = Auxiliar de enfermería
  - 4 = Promotor(a) de salud
  - 5 = Funcionario de registro civil
  - 9 = Sin información


In [ ]:
for col in nac_df:
    print(col, ":", nac_df[col].unique())

ANO : [2017 2018 2021 2022]
AREANAC : [1 3 2 9]
AREA_RES : [ 1.  2.  3. nan  9.]
ATEN_PAR : [1 5 6 2 3 4 9]
CODMUNRE : [  1. 786. 130. 466. 356. 698. 430. 885. 580. 433. 128. 147. 797.  30.
 551. 834. 899. 154. 570. 650. 245.  10. 360. 243. 807. 283. 675. 279.
 573.  51. 215. 399.  79. 111. 860.  32.  45. 172. 276. 126. 238. 667.
 266. 823.  88. 758. 799. 383. 109. 517. 530. 520. 555. 858. 591. 685.
 874. 503. 212. 585. 359.  13. 612. 616. 532.  38. 190. 754. 284. 671.
 837. 438. 175. 516. 268.  91. 621. 250. 357.  34. 151. 873. 101. 307.
 892. 244. 615.  90. 317. 553. 701. 847. 568. 286. 682. 855. 298. 468.
 755. 142. 835. 432. 227. 980. 419. 798.  52. 756.  81. 406. 770. 400.
 417. 688. 320. 181.   2. 318. 865. 380. 376. 590. 483. 377.  58. 794.
 579. 659.  78. 402. 810.  42. 815. 887. 440. 547. 753. 455. 162. 407.
 678. 275. 647. 498. 569. 197. 226. 443. 396. 200. 491. 890. 418. 248.
 290. 490. 240. 773. 421. 222. 234. 545. 170. 548. 836. 364. 330. 662.
 708. 473. 660. 689. 247. 385

Se puede ver que los atributos poseen valores validos salvo por dos de ellos : *ULTCURMAD* y *ULTCURPAD*, ya que registran un valor de 0 cuando este no se encuentra dentro de sus posibles valores

Por ultimo, se verifica que *CODMUNRE* y *COD_MUNIC* tengan valores validos

In [34]:
municipios_df = pd.read_csv("RegionesFormato.csv")
codigosMunicipio = set(municipios_df["CÓDIGO DANE DEL MUNICIPIO"].unique())

municipiosNac = set(nac_df["COD_MUNIC"].unique()).union(set(nac_df["CODMUNRE"].unique()))

municipiosFaltantes = municipiosNac - codigosMunicipio

print(len(municipiosFaltantes))
municipiosFaltantes

0


set()

Vemos que todos los municipios son válidos.

## 2.1.4. Análisis de consistencia



Existen varios problemas de consistencia al tratar a los valores nulos. A continuación se iran mencionando y corrigiendo

- Se reemplazan valores inválidos por el valor correspondiente a "Sin información" en las variables `ULTCURMAD` y `ULTCURPAD`

In [ ]:
nac_df['ULTCURMAD'] = nac_df['ULTCURMAD'].replace(0, 99)
nac_df['ULTCURPAD'] = nac_df['ULTCURPAD'].replace(0, 99)

- Existen valores nulos, pese a que existen atributos para denotar carencia de información

In [ ]:
nac_df['PROFESION'] = nac_df['PROFESION'].fillna(9).astype(int)
nac_df['AREA_RES'] = nac_df['AREA_RES'].fillna(9).astype(int)

- Las variables `CODMUNRE`y `CODPTORE` no cuentan con un valor para "Sin información". Debido a que son pocas las filas que presentan valores nulos, se procederá a eliminarlas.

In [ ]:
prev_len = nac_df.shape[0]
nac_df = nac_df.dropna()
print(f"Se eliminaron {prev_len - nac_df.shape[0]} registros")

Finalmente, se exporta el archivo corregido

In [35]:
nac_df.to_csv("NacProcesado.csv", index=False)

Aquí tienes una propuesta de cambios para tu análisis de estadísticas vitales basado en el modelo de dimensiones y hechos similar al que sugeriste previamente. Este enfoque usa el concepto de un almacén de datos (Data Warehouse), con tablas de dimensiones y hechos, para modelar los datos de nacimiento que tienes.

## **Propuesta de Cambios en las Dimensiones y Hechos**

---

### **Dim_Fecha**
**Fuente de datos**: nac2017, nac2018, nac2021, nac2022

**Campos**:
- ANO (Año de ocurrencia del nacimiento)
- MES (Mes de la ocurrencia del nacimiento)

**Transformaciones**:
1. Renombrar `ANO` a `anio`.
2. Agregar una columna `trimestre` calculada en base a `MES`.
3. Crear la columna `idFecha` combinando `anio` y `mes` en formato `YYYY-MM`.

**Destino de datos**: `Dim_Fecha`

**Campos resultantes**:
- idFecha
- mes
- trimestre
- anio

---

### **Dim_Madre**
**Fuente de datos**: nac2017, nac2018, nac2021, nac2022

**Campos**:
- EDAD_MADRE (Edad de la madre a la fecha del parto)
- EST_CIVM (Estado conyugal de la madre)
- NIV_EDUM (Último nivel de estudio de la madre)
- ULTCURMAD (Último año o grado aprobado por la madre)
- SEG_SOCIAL (Régimen de seguridad social de la madre)

**Transformaciones**:
1. Renombrar `EDAD_MADRE` a `rango_edad`.
2. Renombrar `EST_CIVM` a `estado_conyugal`.
3. Renombrar `NIV_EDUM` a `maximo_nivel_educativo`.
4. Renombrar `ULTCURMAD` a `maximo_año_o_grado_aprobado`.
5. Renombrar `SEG_SOCIAL` a `regimen_seguridad_social`.
6. Reemplazar los valores `0` en `maximo_año_o_grado_aprobado` con `99`.
7. Transformar los valores categóricos en su correspondiente descripción (por ejemplo, 1 = “Contributivo”, etc.).

**Destino de datos**: `Dim_Madre`

**Campos resultantes**:
- rango_edad
- estado_conyugal
- maximo_nivel_educativo
- maximo_año_o_grado_aprobado
- regimen_seguridad_social

---

### **Dim_Hijo**
**Fuente de datos**: nac2017, nac2018, nac2021, nac2022

**Campos**:
- SEXO (Sexo del nacido vivo)
- PESO_NAC (Peso al nacer)

**Transformaciones**:
1. Renombrar `PESO_NAC` a `peso`.
2. Agregar la columna `talla` y establecer su valor como `null` (vacío).
3. Reemplazar los valores nulos en `peso` con `9`.
4. Convertir los valores categóricos en descripciones.

**Destino de datos**: `Dim_Hijo`

**Campos resultantes**:
- sexo
- peso
- talla

---

### **Dim_Parto**
**Fuente de datos**: nac2017, nac2018, nac2021, nac2022

**Campos**:
- TIPO_PARTO (Tipo de parto)
- MUL_PARTO (Multiplicidad del embarazo)

**Transformaciones**:
1. Renombrar `TIPO_PARTO` a `tipo_parto`.
2. Renombrar `MUL_PARTO` a `multiplicidad_embarazo`.
3. Reemplazar los valores categóricos por su descripción correspondiente.
4. Reemplazar los valores `4` en `TIPO_PARTO` con `9` (sin información).

**Destino de datos**: `Dim_Parto`

**Campos resultantes**:
- tipo_parto
- multiplicidad_embarazo

---

### **Dim_Atencion_Parto**
**Fuente de datos**: nac2017, nac2018, nac2021, nac2022

**Campos**:
- SIT_PARTO (Sitio de parto)
- ATEN_PAR (Atención del parto)
- PROFESION (Profesión de quien certifica el parto)

**Transformaciones**:
1. Renombrar `SIT_PARTO` a `sitio_parto`.
2. Renombrar `ATEN_PAR` a `atendido_por`.
3. Renombrar `PROFESION` a `profesion_certificadora`.
4. Reemplazar valores categóricos por descripciones textuales (por ejemplo, 1 = “Médico”).

**Destino de datos**: `Dim_Atencion_Parto`

**Campos resultantes**:
- sitio_parto
- atendido_por
- profesion_certificadora

---

### **Hecho_Nacimiento**
**Fuente de datos**: nac2017, nac2018, nac2021, nac2022

**Campos**:
- N_HIJOSV (Número de hijos nacidos vivos)
- PESO_NAC (Peso al nacer)
- T_GES (Tiempo de gestación)
- T_GES_AGRU_CIE (Tiempo de gestación ajustado CIE)
- MUL_PARTO (Multiplicidad del embarazo)
- TIPO_PARTO (Tipo de parto)

**Transformaciones**:
1. Renombrar `N_HIJOSV` a `num_hijos_nacidos_vivos_madre`.
3. Renombrar `PESO_NAC` a `peso_nacido_vivo`.
4. Renombrar `T_GES` a `tiempo_gestacion`.
5. Renombrar `T_GES_AGRU_CIE` a `tiempo_gestacion_CIE`.
6. Renombrar `MUL_PARTO` a `multiplicidad_embarazo`.
7. Renombrar `TIPO_PARTO` a `tipo_parto`.
8. Reemplazar los valores categóricos en `MUL_PARTO`, `TIPO_PARTO`, `T_GES`, y `T_GES_AGRU_CIE` por descripciones textuales:
   - **MUL_PARTO**: 
     - 1 = Simple
     - 2 = Doble
     - 3 = Triple
     - 4 = Cuádruple o más
     - 9 = Sin información
   - **TIPO_PARTO**: 
     - 1 = Espontáneo
     - 2 = Cesárea
     - 3 = Instrumentado
     - 9 = Sin información
   - **T_GES** y **T_GES_AGRU_CIE**:
     - 1 = Menos de 22 semanas
     - 2 = De 22 a 27 semanas
     - 3 = De 28 a 37 semanas
     - 4 = De 38 a 41 semanas
     - 5 = 42 semanas o más
     - 9 = Sin información

**Destino de datos**: `Hecho_Nacimiento`

**Campos resultantes**:
- num_hijos_nacidos_vivos_madre
- peso_nacido_vivo
- tiempo_gestacion
- tiempo_gestacion_CIE
- multiplicidad_embarazo
- tipo_parto